scripts/validate/shellcheck

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env bash
+#!/usr/bin/env sh
 set -eo pipefail
```

Comment:
Looks like we may need bash after all if we want to keep the `set -eo pipefail`, see https://github.com/docker/cli/pull/1692

```
/go/src/github.com/docker/cli/scripts/build/windows: 6: set: Illegal option -o pipefail
/go/src/github.com/docker/cli/scripts/build/osx: 6: set: Illegal option -o pipefail
```

(although it seems to work fine for some reason, so slightly confused)

Reply:
Looks like busybox sh supports it (but `dash` will fail)

```
dash: 1: set: Illegal option -o pipefail
```